src/common/libsubprocess/subprocess.c

```diff
@@ -546,12 +546,20 @@ flux_subprocess_t * flux_exec (flux_t *h, int flags,
     return flux_exec_wrap (h, r, flags, cmd, ops, hooks);
 }
 
+flux_subprocess_t *flux_local_exec_ex (flux_reactor_t *r,
```

Comment:
> Problem: flux_local_exec() takes a "hooks" parameter that is rarely used.

I only see 4 places (besides tests) that use `flux_local_exec()`. Two use the hooks parameter and two do not. I may be missing something that makes half the usage "rarely used", but is the churn here really worth it? Most of the change is introduced in the unit tests, but maybe this change makes some future changes cleaner?

Reply:
Ah, maybe this was necessary to add a function that takes the llog parameters anyway... (now that I'm viewing the next commit)

Reply:
That's a fair point.  I guess it would have been fine to leave the hooks parameter in `flux_local_exec()` and my claim of rarity is on thin ice.  Should I move it back there?

Reply:
Just pushed with typo fixed and a more honest claim in the commit that creates `flux_local_exec_ex()`.  I could easily move `hooks` back to flux_local_exec() if that's better.

Reply:
Nah, after seeing the other ex-tra parameters added to `flux_local_exec_ex()`, it seems fine to me!